wezterm-gui/src/termwindow/mod.rs

```diff
@@ -1018,44 +1018,124 @@ impl TermWindow {
                 Ok(true)
             }
             WindowEvent::DroppedUrl(urls) => {
                 let pane = match self.get_active_pane_or_overlay() {
                     Some(pane) => pane,
                     None => return Ok(true),
                 };
-                let urls = urls
-                    .iter()
-                    .map(|url| self.config.quote_dropped_files.escape(&url.to_string()))
-                    .collect::<Vec<_>>()
-                    .join(" ")
-                    + " ";
-                pane.send_paste(urls.as_str())?;
+                let urls_string = self.user_dropped_event( 
+                    "user-dropped-urls",
+                    urls.iter().map(|url| url.to_string()).collect::<Vec<_>>(),
+                );
+                pane.send_paste(
+                    urls_string.as_deref().unwrap_or(
+                        &urls
+                            .iter()
+                            .map(|url| url.to_string())
+                            .collect::<Vec<_>>()
+                            .join(" "),
+                    ),
+                )?;
                 Ok(true)
             }
             WindowEvent::DroppedFile(paths) => {
                 let pane = match self.get_active_pane_or_overlay() {
                     Some(pane) => pane,
                     None => return Ok(true),
                 };
-                let paths = paths
-                    .iter()
-                    .map(|path| {
-                        self.config
-                            .quote_dropped_files
-                            .escape(&path.to_string_lossy())
-                    })
-                    .collect::<Vec<_>>()
-                    .join(" ")
-                    + " ";
-                pane.send_paste(&paths)?;
+                let paths_string = self.user_dropped_event( 
+                    "user-dropped-paths",
+                    paths
+                        .iter()
+                        .map(|path| path.to_string_lossy().to_string())
+                        .collect::<Vec<_>>(),
+                );
+                pane.send_paste(
+                    paths_string.as_deref().unwrap_or(
+                        (paths
+                            .iter()
+                            .map(|path| {
+                                self.config
+                                    .quote_dropped_files
+                                    .escape(&path.to_string_lossy())
+                            })
+                            .collect::<Vec<_>>()
+                            .join(" ")
+                            + " ")
+                            .as_str(),
+                    ),
+                )?;
                 Ok(true)
             }
             WindowEvent::DraggedFile(_) => Ok(true),
         }
     }
+    fn user_dropped_event(
+        &mut self, 
+        event_name: &str, 
+        data: Vec<String>
+    ) -> Option<String> {
```

Comment:
I think there are three cases that need to be defined:

* The event provided the string (by returning the string)
* The event wants the default processing to occur (`return nil` or not explicitly returning anything)
* The event wants nothing to be sent. Return an empty string? Something better?

Reply:
I believe this is what I have implemented. 

The user can return an empty string to cause no further events to go. This would allow the user to even use `pane:send_text` rather than `pane:send_paste` if wanted. (as my examples even show in docs)